util.go

```diff
@@ -218,3 +219,18 @@ func (bg *blockGetter) LogStats() {
 }
 
 var _ ipldcbor.IpldBlockstore = &blockGetter{}
+
+type filRpcBs struct {
+	rpc fil.LotusDaemonAPIClientV0
+}
+
+func (frbs *filRpcBs) Put(context.Context, blkfmt.Block) error {
+	return xerrors.New("this is a readonly store")
+}
+func (rbs *filRpcBs) Get(ctx context.Context, c cid.Cid) (blkfmt.Block, error) {
+	d, err := rbs.rpc.ChainReadObj(ctx, c)
```

Comment:
Out of curiosity is this more performant than just peering with the libp2p host within lotus?

Seems like they should be pretty similar, but wouldn't be surprised if say lotus' bitswap server had more rate-limiting than the API endpoint which made the API faster.

No objection if this is faster/easier, but wanted to verify since if bitswap works just as well it's fewer dependencies and it would add nicely as an additional data source.

Reply:
Thanks for the info. Unfortunate, but not so surprising given the rate-limiting. Were these nodes on the same machine or was there some latency?

IIRC the rate-limiting used in boxo (and lotus) is more prominent the higher the latency between peers.